selfdrive/ui/qt/offroad/networking.cc

```diff
@@ -169,12 +168,21 @@ AdvancedNetworking::AdvancedNetworking(QWidget* parent, WifiManager* wifi): QWid
     } else {
       params.put("GsmApn", apn.toStdString());
     }
-    wifi->updateGsmSettings(roamingEnabled, apn);
+    wifi->updateGsmSettings(params.getBool("GsmRoaming"), apn, params.getBool("GsmUnmetered"));
   });
   list->addItem(editApnButton);
 
+  // Metered toggle
+  const bool unmeteredOverride = params.getBool("GsmUnmetered");
+  ToggleControl *meteredToggle = new ToggleControl(tr("Force Unmetered Cellular"), "", "", !unmeteredOverride);
```

Comment:
this is slightly confusing; I would simplify the toggle to just say "cellular metered" (but keep using no and unknown behind the hood)

Reply:
let's also add a description

Reply:
how is `Prevent large data uploads when on a metered connection`?

Reply:
lgtm. let's make sure it also defaults to metered/unknown

Reply:
does this only apply to cellular? sounds like it might apply to mobile hotspots

Reply:
oh, what about `Prevent large data uploads on cellular connection`?